### ISSUE TITLE
Support more than 2 Istio HTTPRouteDestinations in the Canary rollout

### ISSUE THREAD

**Issue by wentwog** [argoproj/argo-rollouts#1364]
Currently the Canary rollout requires that Istio VirtualService's HTTPRoute must have exactly 2 HTTPRouteDestinations.
Support having more HTTPRouteDestinations.

# Use Cases

Migrating to Istio, we have a legacy system to which we route some percentage of traffic, e.g.:
- Istio Stable: 50%
- Istio Canary: 30%
- Legacy: 20%

When a new service version gets rolled out, we'd like ArgoRollouts to switch the weights only for the first two HTTPRouteDestinations.
Note, Stable + Canary weight might not be 100% in this case.
The Legacy weight is changed manually.

---
**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

**Comment by pparth** [comment]
It is rather restrictive to have to use exactly 2 Route Destinations, unless we are missing something.

**Comment by agrawroh** [comment]
Since I already touched some of the code for Argo integration with Istio VS in one of my other PRs, I could work on this one. We have a similar requirement and it'll be pretty cool to get it done.
cc @huikang

**Comment by pparth** [comment]
So, is there an ETA about this?

**Comment by agrawroh** [comment]
Created a PR for this:
https://github.com/argoproj/argo-rollouts/pull/1420

Would appreciate any feedback/comments.

**Comment by angeloslenis** [comment]
Any updates here?

**Comment by jonwinton** [comment]
Hello! This is an issue I've come across that I'm interested in solving for our org. @agrawroh I saw from your PR that there was a question about the approach to the design and [you had a suggestion](https://github.com/argoproj/argo-rollouts/pull/1420#discussion_r690441316):

> One proposal is to add an optional httpRoutes which will be an array of objects similar to the tlsRoutes and gradually deprecate the use of routes over time. In the beginning, we can enforce the use of only one i.e either routes or httpRoutes. I proposed this to @jessesuen as well.

To pick this up, would you recommend following this route? Or do you think the approach of `maxWeight` is the appropriate?

**Comment by agrawroh** [comment]
> Hello! This is an issue I've come across that I'm interested in solving for our org. @agrawroh I saw from your PR that there was a question about the approach to the design and [you had a suggestion](https://github.com/argoproj/argo-rollouts/pull/1420#discussion_r690441316):
> 
> > One proposal is to add an optional httpRoutes which will be an array of objects similar to the tlsRoutes and gradually deprecate the use of routes over time. In the beginning, we can enforce the use of only one i.e either routes or httpRoutes. I proposed this to @jessesuen as well.
> 
> To pick this up, would you recommend following this route? Or do you think the approach of `maxWeight` is the appropriate?

Yes, that's probably the right way to do it. If you want to give it a shot, I'm more than happy to review :)
